pkg/search/postgres/store_cache.go

```diff
@@ -237,7 +238,9 @@ func (c *cachedStore[T, PT]) Exists(ctx context.Context, id string) (bool, error
 
 // Count returns the number of objects in the store matching the query.
 func (c *cachedStore[T, PT]) Count(ctx context.Context, q *v1.Query) (int, error) {
-	if q == nil || q.EqualVT(search.EmptyQuery()) {
+	// Check scope queries
+	scopeQuery, err := scoped.GetQueryForAllScopes(ctx)
+	if (scopeQuery == nil || err != nil) && (q == nil || q.EqualVT(search.EmptyQuery())) {
```

Comment:
if `GetQueryForAllScopes` returns an error, we go straight to cache without honoring scopes but without logging that an error occurred hence the query results may be incorrect since we have just given you the results without considering scope?